nginx.md

```diff
@@ -0,0 +1,190 @@
+
+## Getting Started
+
+While the [example]({{ site.github.url }}/docs#nginx-configuration) provided on the previous page is a good starting point, it will likely not be enough to fully configure your Pico installation.
+
+There are three main parts to configuring a Pico site under Nginx, in addition to general server configuration.  The three sets of rules we will be developing provide the following functions: Denying access to Pico's internal files, configuring PHP, and setting up Pico's url-rewriting.  Although it's arguably the most important function, we'll be configuring url-rewriting last due to the order that Nginx processes its config.  We'll discuss that in more detail when we get to it below.
+
+## General Server Configuration
+
+Nginx's server configuration is broken into sections called `blocks`.  Each website has its own `server` block inside your Nginx config.  While configuration of Nginx sites, virtual hosts, and other aspects of this topic are outside the scope of this guide, we'll provide enough to at least get you started with Pico.
+
+```
+server {
+	listen 80;
+
+	server_name www.example.com example.com;
+	root /var/www/example;
+
+	index index.php;
+}
+```
+
+Let's break down this example.  The first line, `listen 80` tells Nginx to listen on port 80 for incoming connections.  This is the default port used by web traffic, and what you'll want to use 99% of the time.  `server_name` is where you specify what domain name or names match this configuration.  You'll likely want to include your domain both with and without the `www.` subdomain.  `root` lets you specify the Document Root for this site.  This is usually going to be where you've installed Pico, but ultimately it depends on your configuration.  `index index.php` tells Nginx that your site's index file will be called `index.php`.  This is necessary for Pico, however you can use something like `index index.html index.htm index.php` if you'd like Nginx to also search for a non-Pico html index file.
+
+Below we'll add a few more sections to this server configuration.  All of the following examples should be placed **inside** the server block, before the closing `}`
+
+## Denying Access to Pico's Internal Files
+
+One of the important features provided by Pico's `.htaccess` is the denial of access to some of Pico's directories.  There are two simple reasons for this, added security, and ease-of-use.  We don't want anyone snooping around and reading files they shouldn't be (like our Pico `config.php`), but we also don't want people accidentally navigating to a raw markdown file or viewing directory indexes (not actually an issue, since they're disabled unless specified in Nginx, but this isn't always the case in Apache).
+
+You should always use the following rule in the Nginx config of your Pico site:
+
+```
+location ~ /(\.htaccess|\.git|config|content|content-sample|lib|vendor) {
```

Comment:
This rule also affects non-Pico subdirectories, isn't it? Especially blocking access to `content` globally may cause trouble with other software. Does Nginx allow cascading `location` blocks? Otherwise a users needs to adjust the base path (i.e. `/` or `/pico/`) manually :unamused:

In general: Shouldn't this section come after the PHP configuration or is this necessary because of Nginx's processing order? The document should follow a golden thread from "General Configuration" to "Pico's configuration" - at least as long as Nginx's configuration order doesn't foil it.